File: apprefurbished/src/main/java/com/androidcollider/easyfin/transaction_categories/root/TransactionCategoriesRootModule.java
package com.androidcollider.easyfin.transaction_categories.root;

import android.content.Context;

import com.androidcollider.easyfin.common.repository.Repository;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ihor Bilous
 */

@Module
public class TransactionCategoriesRootModule {

    @Provides
    TransactionCategoriesRootMVP.Model provideTransactionCategoriesRootMVPModel(Repository repository) {
        return new TransactionCategoriesRootModel(repository);
    }

    @Provides
    TransactionCategoriesRootMVP.Presenter provideTransactionCategoriesRootMVPPresenter(Context context,
                                                                                         TransactionCategoriesRootMVP.Model model) {
        return new TransactionCategoriesRootPresenter(context, model);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/transaction_categories/root/TransactionCategoriesRootModule.kt
package com.androidcollider.easyfin.transaction_categories.root

import android.content.Context
import com.androidcollider.easyfin.common.repository.Repository
import dagger.Module
import dagger.Provides

/**
 * @author Ihor Bilous
 */
@Module
class TransactionCategoriesRootModule {
    @Provides
    fun provideTransactionCategoriesRootMVPModel(repository: Repository?): TransactionCategoriesRootMVP.Model {
        return TransactionCategoriesRootModel(repository)
    }

    @Provides
    fun provideTransactionCategoriesRootMVPPresenter(
        context: Context?,
        model: TransactionCategoriesRootMVP.Model?
    ): TransactionCategoriesRootMVP.Presenter {
        return TransactionCategoriesRootPresenter(context, model)
    }
}